collectors/modality-probe-collector-common/src/lib.rs

```diff
@@ -37,78 +42,52 @@ newtype! {
 }
 
 newtype! {
-    /// A log segment
-    ///
-    /// The log is divided into segments, each of which begins with some logical
-    /// clock entries and ends with a sequence of events. The id must be unique
-    /// within the session.
+    /// A log report sequence number
     #[derive(Debug, Eq, PartialEq, Hash, Copy, Clone)]
-    pub struct SegmentId(pub u32);
-}
-
-/// Map an event id to its name and description
-#[derive(Debug, Eq, PartialEq)]
-pub struct EventMapping {
-    id: EventId,
-    name: String,
-    description: String,
-}
-
-/// Map a probe id to its name and description
-#[derive(Debug, Eq, PartialEq)]
-pub struct ProbeMapping {
-    id: ProbeId,
-    name: String,
-    description: String,
+    pub struct SequenceNumber(pub u64);
 }
 
-/// The data that may be attached to a log entry
-#[derive(Debug, Eq, PartialEq, Clone)]
-pub enum LogEntryData {
-    Event(EventId),
-    EventWithPayload(EventId, u32),
-    LogicalClock(ProbeId, ProbeEpoch, ProbeTicks),
-}
+#[derive(Debug)]
+pub struct SerializationError;
```

Comment:
This seems like it would be a maddening error to receive, since there's no context whatsoever and greater than 5 ways it might be triggered.

Reply:
Oh, for sure. There was once a TODO here to fill this in and add error derivation. It must have gotten eaten in the shuffle.

Reply:
I fixed this locally; will push it shortly.